File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/activity/MainActivity.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.activity;

import android.os.Bundle;

import com.github.rooneyandshadows.lightbulb.application.activity.BaseActivity;
import com.github.rooneyandshadows.lightbulb.application.activity.routing.BaseApplicationRouter;

import androidx.annotation.Nullable;

public class MainActivity extends BaseActivity {
    private AppRouter router;

    @Nullable
    @Override
    protected BaseApplicationRouter initializeRouter(int fragmentContainerId) {
        router = new AppRouter(this, fragmentContainerId);
        Router.getInstance().setRouter(router);
        return router;
    }

    @Override
    protected void create(@Nullable Bundle savedInstanceState) {
        if (savedInstanceState == null)
            router.toDemoRegular(BaseApplicationRouter.NavigationCommands.NAVIGATE_TO_AND_CLEAR_BACKSTACK);
    }
}

File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/activity/MainActivity.kt
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.activity

import android.os.Bundle
import com.github.rooneyandshadows.lightbulb.application.activity.BaseActivity
import com.github.rooneyandshadows.lightbulb.application.activity.slidermenu.config.SliderMenuConfiguration
import com.github.rooneyandshadows.lightbulb.apt.annotations.LightbulbActivity
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.R
import com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.lightbulb.service.LightbulbService

@LightbulbActivity(fragmentContainerId = "fragmentContainer")
class MainActivity : BaseActivity() {

    override fun doBeforeCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.DemoTheme)
    }

    override fun getMenuConfiguration(): SliderMenuConfiguration {
        return MenuConfigurations.getConfiguration(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (savedInstanceState == null) {
            LightbulbService.route().toDemoRegular().newRootScreen()
        }
    }

    override fun onUnhandledException(paramThread: Thread?, exception: Throwable) {
        super.onUnhandledException(paramThread, exception)
        exception.printStackTrace()
    }
}